third_party/move/move-compiler-v2/src/pipeline/ability_checker.rs

```diff
@@ -3,10 +3,10 @@
 
 //! Checks for ability violations.
 //! prerequisite:
-//! - liveness analysis and lifetime analysis have been performed
 //! - Copies and moves have been made explicit in assignment instructions
 
-use move_binary_format::file_format::{Ability, AbilitySet};
+use crate::pipeline::abort_analysis::{ExitStateAnnotation, ExitStateAtCodeOffset};
```

Comment:
At this point I have a pretty extensive rewrite of this module in place. (See also message on slack.) We can land it now, or remove the changes from this PR, I just need the abort analysis.